### PR TITLE
HTCONDOR-2309: New Python bindings, part 4.

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/python-bindings/htcondor2/schedd.cpp [hunk 1]
```diff
@@ -614,14 +614,26 @@ submitProcAds( int clusterID, long count, SubmitBlob * sb, ClassAd * & clusterAd
...
             return -1;
         }
 
+        // If we're spooling, we'll need a copy of each proc ad later,
+        // so just store them now, rather than try to recreate them.
+        // Don't chain them to this clusterAd; the Python SubmitResult
+        // object will be giving us a copy of it when we need it.
+        if( spooledProcAds ) {
+            ClassAd * copy = new ClassAd( * procAd );
+            copy->Assign( ATTR_CLUSTER_ID, clusterID );
```
Comment (johnkn) [inline src/python-bindings/htcondor2/schedd.cpp:631]: proc ads should not have a cluster id attribute.  probably not a problem in practice though...

**Review by johnkn — APPROVED** [review]
Approved pending (possible?) change from comment.

File: src/python-bindings/htcondor2/schedd.cpp (hunk 1 above)
Comment (Todd-L-Miller) [reply src/python-bindings/htcondor2/schedd.cpp:631]: For the record, I just I cargo-culted this in from https://github.com/htcondor/htcondor/blob/3997589a06f2bdda66a63260f49ecbf8e0b9853a/src/condor_submit.V6/submit.cpp#L2454 :)

**Comment by Todd-L-Miller** [comment]
We discussed this.  It's harmless but unnecessary.

**Pull request merged**